libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -372,6 +374,10 @@ bool AP_Mission::set_current_cmd(uint16_t index)
         if (is_nav_cmd(cmd)) {
             // save previous nav command index
             _prev_nav_cmd_index = _nav_cmd.index;
+            // save separate previous nav command index if it contains lat,long,alt
+            if (cmd.content.location.lat != 0 && cmd.content.location.lng != 0) {
```

Comment:
i think this should be an || not an && (unless you have some prejudice against people landing on the equator?)